Sources/UtilityBeltData/CoreData/CoreDataManager.swift

```diff
@@ -0,0 +1,157 @@
+// Copyright © 2020 SpotHero, Inc. All rights reserved.
+
+import CoreData
+import Foundation
+
+/// A helper class to assist with all CoreData operations.
+public class CoreDataManager {
+    // MARK: - Default Instance
+
+    /// The default instance of CoreDataManager.
+    /// Until the default container is set, any operations on this instance will fail due to missing context.
+    /// This is initialized with an empty persistent container
+    public private(set) static var `default` = CoreDataManager()
+
+    /// The default persistent container to use when referencing the default instance of CoreDataManager.
+    public static var defaultContainer: NSPersistentContainer? {
+        get {
+            return self.default.persistentContainer
+        }
+        set {
+            self.default.persistentContainer = newValue
+        }
+    }
+
+    // MARK: - Properties
+
+    /// The persistent container to use for all requests. Must be set prior to performing any operations.
+    private var persistentContainer: NSPersistentContainer?
+
+    // MARK: - Methods
+
+    // MARK: Initializers
+
+    /// Initializes a new CoreDataManger with no persistent container.
+    /// This is only used for the static default instance of this class.
+    /// All operations on the default instance will fail until the default container is set.
+    private init() {}
+
+    /// Initializes a new CoreDataManager.
+    /// - Parameter container: The persistent container to use for all operations.
+    public required init(container: NSPersistentContainer) {
```

Comment:
why do we have a `default` CoreDataManager that won't work until we set `defaultContainer`, and then also have this initializer?  Whats the point of having a default manager with no defaultContainer, and if you have both, why wouldnt you just use this initializer? just trying to understand this pattern better.

Reply:
@bdrelling any thoughts here? does what im asking make sense?

Reply:
I think this is me trying to allow it to be a singleton to match the format we currently have in our project, such that you'd set `CoreDataManager.default.persistentContainer = ...` before any requests are made, but at the same time, I'd almost rather we avoid making this a singleton and our app can create a singleton instance if it wants to that uses some kind of `CoreDataOperator` class or something. Does that make more sense?